src/main/groovy/org/shipkit/internal/gradle/git/GitAuthPlugin.java

```diff
@@ -0,0 +1,113 @@
+package org.shipkit.internal.gradle.git;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.shipkit.gradle.ReleaseConfiguration;
+import org.shipkit.internal.exec.DefaultProcessRunner;
+import org.shipkit.internal.gradle.configuration.ReleaseConfigurationPlugin;
+import org.shipkit.internal.notes.vcs.GitOriginRepoProvider;
+
+import java.text.MessageFormat;
+
+public class GitAuthPlugin implements Plugin<Project>{
+
+    private static final Logger LOG = Logging.getLogger(GitAuthPlugin.class);
+
+    private GitAuth gitAuth;
+
+    @Override
+    public void apply(Project project) {
+        ReleaseConfiguration conf = project.getPlugins().apply(ReleaseConfigurationPlugin.class).getConfiguration();
+        String ghUser = conf.getGitHub().getWriteAuthUser();
+        String writeToken = conf.getGitHub().getWriteAuthToken();
+
+        String configUrl;
+        String originUrl;
+        String secretValue = null;
+        String originRepositoryName = getOriginRepositoryName(project);
+        if (writeToken != null) {
+            LOG.lifecycle("  'git push/pull' use GitHub write token.");
+            configUrl = getUrlWithAuth(ghUser, conf.getGitHub().getRepository(), writeToken);
+            originUrl = getUrlWithAuth(ghUser, originRepositoryName, writeToken);
+            secretValue = writeToken;
+        } else {
+            LOG.lifecycle("  'git push/pull' do not use GitHub write token because it was not specified.");
+            configUrl = getUrl(conf.getGitHub().getRepository());
+            originUrl = getUrl(originRepositoryName);
+        }
+        gitAuth = new GitAuth(configUrl, originUrl, secretValue, originRepositoryName);
+    }
+
+    private String getUrl(String ghRepo) {
+        return MessageFormat.format("https://github.com/{0}.git", ghRepo);
+    }
+
+    private String getOriginRepositoryName(Project project) {
+        DefaultProcessRunner runner = new DefaultProcessRunner(project.getProjectDir());
+        GitOriginRepoProvider repoProvider = new GitOriginRepoProvider(runner);
+        try{
+            return repoProvider.getOriginGitRepo();
+        } catch(Exception e){
+            LOG.debug("Failed to get local git remote origin", e);
```

Comment:
I would probably do "info" level message and tweak it a bit (we should not use keywords like "failed" or "error" at the info level - they are very confusing when there are other real errors). What is the consequence if the origin git repo is null? Will the task fail saying that @Input is not satisfied? The best developer experience is when failures are easy to figure out :)